### PR TITLE
fix pybind11 incompatibility with some CUDA versions and add Catch2 dependency for tests

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
taurusml7 - Linux RHEL 7.6, POWER, 8335-GTX, 6 x NVIDIA Tesla V100-SXM2-32GB, 440.64.00, Python 2.7.5
See https://gist.github.com/Flamefire/1f0b35ad2996c3438d06758a4542be6d for a full test report.

File: easybuild/easyconfigs/p/pybind11/pybind11-2.10.3-GCCcore-12.2.0.eb [hunk 1]
```diff
@@ -9,15 +9,24 @@ toolchain = {'name': 'GCCcore', 'version': '12.2.0'}
...
 builddependencies = [
     ('binutils', '2.39'),
     ('CMake', '3.24.3'),
+    # Test dependencies
     ('Eigen', '3.4.0'),
+    ('Catch2', '2.13.9', '', SYSTEM),
 ]
-dependencies = [('Python', '3.10.8')]
 
-configopts = "-DPYTHON_EXECUTABLE=$EBROOTPYTHON/bin/python"
```
Comment (Micket) [inline easybuild/easyconfigs/p/pybind11/pybind11-2.10.3-GCCcore-12.2.0.eb:21]: was moving this line completely from only python11-2.10.3 intentional?

Comment (Flamefire) [reply easybuild/easyconfigs/p/pybind11/pybind11-2.10.3-GCCcore-12.2.0.eb:21]: It was supposed to get removed from both, but missed the 2nd. This is done in the easyblock so we don't need to copy this around to all ECs. Pushed an update

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen2

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/022c73dd80f9e156c099ebf9102823a6 for a full test report.

**Comment by Micket** [comment]
Going in, thanks @Flamefire!

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
taurusi8006 - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor, 8 x NVIDIA NVIDIA A100-SXM4-40GB, 470.57.02, Python 2.7.5
See https://gist.github.com/Flamefire/069cb70678a1a5b30a68799482f1342d for a full test report.

**Pull request merged**